Gems/ROS2/Code/Source/RobotImporter/xacro/XacroUtils.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "XacroUtils.h"
+#include <AzCore/IO/FileIO.h>
+#include <AzCore/XML/rapidxml.h>
+#include <QProcess>
+#include <QString>
+
+namespace ROS2::Utils::xacro
+{
+
+    ExecutionOutcome ParseXacro(const AZStd::string& filename, const Params& params)
+    {
+        ExecutionOutcome outcome;
+        // test if xacro exists
+        AZStd::string xacroPath = WhichXacro();
+
+        AZ_Warning("ParseXacro", !xacroPath.empty(), "There is xacro executable in your path");
+        if (xacroPath.empty())
+        {
+            return ExecutionOutcome{};
+        }
+        AZ_Printf("ParseXacro", "xacro executable : %s", xacroPath.c_str());
+        AZ_Printf("ParseXacro", "Convert xacro file : %s", filename.c_str());
```

Comment:
repeat: these printfs will all be on the same line.
A newline needs to be added.
```suggestion
        AZ_Printf("ParseXacro", "xacro executable : %s\n", xacroPath.c_str());
        AZ_Printf("ParseXacro", "Convert xacro file : %s\n", filename.c_str());
```